go-controller/pkg/util/gateway-cleanup.go

```diff
@@ -34,34 +36,23 @@ func GatewayCleanup(nodeName string) error {
 			routerIP = strings.Split(routerIPNetwork, "/")[0]
 		}
 	}
-
 	if routerIP != "" {
-		// Get a list of all the routes in cluster router with this gateway
-		// Router as the next hop.
-		var uuids string
-		uuids, stderr, err = RunOVNNbctl("--data=bare", "--no-heading",
-			"--columns=_uuid", "find", "logical_router_static_route",
-			"nexthop="+routerIP)
-		if err != nil {
-			return fmt.Errorf("Failed to fetch all routes with gateway "+
-				"router %s as nexthop, stderr: %q, "+
-				"error: %v", gatewayRouter, stderr, err)
-		}
-
-		// Remove all the routes in cluster router with this gateway Router
-		// as the nexthop.
-		routes := strings.Fields(uuids)
-		for _, route := range routes {
-			_, stderr, err = RunOVNNbctl("--if-exists", "remove",
-				"logical_router", clusterRouter, "static_routes", route)
-			if err != nil {
-				logrus.Errorf("Failed to delete static route %s"+
-					", stderr: %q, err = %v", route, stderr, err)
-				continue
-			}
-		}
+		nextHops = append(nextHops, routerIP)
+	}
+	ip, _, err := net.ParseCIDR(nodeSubnet)
+	if err != nil {
+		return fmt.Errorf("Failed to parse local subnet %s: %v", nodeSubnet, err)
+	}
+	//Get the fixed second IP for the management Port
+	ip = NextIP(NextIP(ip))
+	mgtPortIP = ip.String()
+	if mgtPortIP != "" {
```

Comment:
perhaps, you need to move L42 to 48 to its own function called GetMgmtPortIP(nodeSubnet string) {} function?

Reply:
Done.